sw/device/tests/dif_gpio/dif_gpio_test.cc

```diff
@@ -0,0 +1,407 @@
+    };
+    write_register_fptr = [](dif_gpio_t const *const gpio,
+                             uint32_t const offset, uint32_t const val) {
+      DifGpioTest::mock->write_register(gpio, offset, val);
+    };
+    write_register_masked_access_fptr =
+        [](dif_gpio_t const *const gpio, uint32_t const lowerOffset,
+           uint32_t const upperOffset, uint32_t const mask,
+           uint32_t const val) {
+          DifGpioTest::mock->write_register_masked_access(
+              gpio, lowerOffset, upperOffset, mask, val);
+        };
+    write_register_bit_masked_access_fptr =
+        [](dif_gpio_t const *const gpio, uint32_t const lowerOffset,
+           uint32_t const upperOffset, uint32_t const index, bool const val) {
+          DifGpioTest::mock->write_register_bit_masked_access(
+              gpio, lowerOffset, upperOffset, index, val);
+        };
+  }
+
+  // Helper methods to reduce repetition
+  auto &EXPECT_INDEX_TO_MASK(uint32_t const index) {
+    return EXPECT_CALL(*mock, index_to_mask(index));
+  }
+
+  auto &EXPECT_ABORT_IF_INVALID() {
+    return EXPECT_CALL(*mock, abort_if_invalid(&gpio));
+  }
+
+  auto &EXPECT_REG32_WRITE(uint32_t const offset, uint32_t const val) {
+    return EXPECT_CALL(*mock, reg32_write(gpio.base_addr, offset, val));
+  }
+
+  auto &EXPECT_REG32_READ(uint32_t const offset) {
+    return EXPECT_CALL(*mock, reg32_read(gpio.base_addr, offset));
+  }
+
+  auto &EXPECT_READ_REGISTER(uint32_t const offset) {
+    return EXPECT_CALL(*mock, read_register(&gpio, offset));
+  }
+
+  auto &EXPECT_READ_REGISTER_BIT(uint32_t const offset, uint32_t const index) {
+    return EXPECT_CALL(*mock, read_register_bit(&gpio, offset, index));
+  }
+
+  auto &EXPECT_WRITE_REGISTER(uint32_t const offset, uint32_t const val) {
+    return EXPECT_CALL(*mock, write_register(&gpio, offset, val));
+  }
+
+  auto &EXPECT_WRITE_REGISTER_MASKED_ACCESS(uint32_t const lowerOffset,
+                                            uint32_t const upperOffset,
+                                            uint32_t const mask,
+                                            uint32_t const val) {
+    return EXPECT_CALL(*mock, write_register_masked_access(
+                                  &gpio, lowerOffset, upperOffset, mask, val));
+  }
+
+  auto &EXPECT_WRITE_REGISTER_BIT_MASKED_ACCESS(uint32_t const lowerOffset,
+                                                uint32_t const upperOffset,
+                                                uint32_t const index,
+                                                bool const val) {
+    return EXPECT_CALL(*mock, write_register_bit_masked_access(
+                                  &gpio, lowerOffset, upperOffset, index, val));
+  }
+
+  void TearDown() override {
+    if (mock) {
+      mock.reset();
+    }
+  }
+
+  // Static mock instance
+  static std::unique_ptr<DifGpioMock> mock;
+};
+
+// Initialize static member
+std::unique_ptr<DifGpioMock> DifGpioTest::mock{};
```

Comment:
You can perform initialization in-line.

Reply:
Not really because `mock` is not a literal type.